hack/e2e-kind-cluster-setup.sh

```diff
@@ -25,18 +30,47 @@ setup_cluster() {
 push_local_image() {
     OCI_BIN="$OCI_BIN" IMAGE_REGISTRY="$IMG_REGISTRY" IMAGE_TAG="$IMG_TAG" make manifests
     OCI_BIN="$OCI_BIN" IMAGE_REGISTRY="$IMG_REGISTRY" IMAGE_TAG="$IMG_TAG" make img-build
-    kind load docker-image $IMG_REGISTRY/multus-dynamic-networks-controller:$IMG_TAG
+
+    if [[ $CLUSTER_TYPE == "kind" ]]; then
+      kind load docker-image $IMG_REGISTRY/multus-dynamic-networks-controller:$IMG_TAG
+    else
+      "$OCI_BIN" push $IMG_REGISTRY/multus-dynamic-networks-controller:$IMG_TAG 
+    fi
 }
 
 cleanup() {
     rm -rf multus-cni
     git checkout -- manifests/
 }
 
+install_multus() {
+  if [[ $MULTUS_VERSION == "latest" ]]; then
+    echo "error: MULTUS_VERSION cant be latest, when using non default CLUSTER_TYPE $CLUSTER_TYPE"
```

Comment:
maybe be explicit about what cluster type you need to use here.

Also, why is this limitation in place ?

Reply:
> maybe be explicit about what cluster type you need to use here.

not CLUSTER_TYPE need hint, because for all non default cluster_type cant use the multus setup script
as it also creates a kind cluster (this answers the 2nd question)

there is usage example on PR desc, is it fine ?

Reply:
We reach here only when `CLUSTER_TYPE`  !=  `kind` which is the default value.
So the `CLUSTER_TYPE` is already set correctly, what is not set is `MULTUS_VERSION` and this what the comment says.

The reason we have it, is that for default CLUSTER_TYPE we use `setup_cluster.sh` [1] that creates a kind cluster and then build + deploy latest multus-cni.
Here we dont want to create a new cluster, so for now i just did that we support deploying an official multus release if desired, by providing its `MULTUS_VERSION`.

If we would need to also support latest freshly built multus-cni in the future we can but it isnt as straight forward, started MVP.

[1] https://github.com/k8snetworkplumbingwg/multus-cni/blob/adfb270991d410417a71755d01f53445510ef2a0/e2e/setup_cluster.sh#L25

PR desc has suggested exports.

Reply:
OK, so change two things in the PR:
- stop saying "default cluster type". Start saying kind cluster. Be explicit. 
- explain why: when kind is used, it already has the latest multus deployed.

Reply:
darn. right. I hate bash.